src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java

```diff
@@ -28,12 +35,27 @@
 import uk.gov.cslearning.catalogue.service.RegistryService;
 import uk.gov.cslearning.catalogue.service.upload.AudienceService;
 
-import java.security.Principal;
-import java.util.*;
-import java.util.stream.Collectors;
-
-import static org.springframework.http.HttpStatus.*;
-import static uk.gov.cslearning.catalogue.exception.ResourceNotFoundException.resourceNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
```

Comment:
Can we collapse some of these using statements with wildcard imports without conflict? e.g. `import org.springframework.web.bind.annotation.*`

Reply:
I would not recommend this. Importing with "*" causes all libraries to be imported and compiled. It looks neat in code but is actually not memory efficient.

Reply:
Yes, at compile time, but once the linker runs and determines there's no actual reference to the other imported classes they won't be included in the generated bytecode, surely? (This conversation is a bit academic so to be clear I'm not interested in blocking merge over it)